src/main/java/carpettisaddition/mixins/carpet/tweaks/command/playerActionEnhanced/EntityPlayerActionPackActionTypeUseMixin.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of the Carpet TIS Addition project, licensed under the
+ * GNU Lesser General Public License v3.0
+ *
+ * Copyright (C) 2023  Fallen_Breath and contributors
+ *
+ * Carpet TIS Addition is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Carpet TIS Addition is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Carpet TIS Addition.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package carpettisaddition.mixins.carpet.tweaks.command.playerActionEnhanced;
+
+//#if MC > 11903
+//$$ import carpet.script.utils.Tracer;
+//#else
+import carpet.helpers.Tracer;
+//#endif
+import carpettisaddition.helpers.carpet.playerActionEnhanced.IEntityPlayerActionPackActionTypeUse;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.hit.HitResult;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(targets = "carpet.helpers.EntityPlayerActionPack$ActionType$1")
+public class EntityPlayerActionPackActionTypeUseMixin implements IEntityPlayerActionPackActionTypeUse
+{
+	private float tickPart = 1.0f;
+
+	@Redirect(
```

Comment:
`@Redirect` is bad, try use `@ModifyExpressionValue` or `@WrapOperation` from mixin extra, or set `require = 0` if you do need an `@Redirect`

Reply:
Yes, bad I know. It is carpet code at least, not MC. Added the `require = 0`, but I was unable to get rid of it, @ModifyExpressionValue gave an unhelpful error message, but I assume the static function is the problem.